src/rmodels.c

```diff
@@ -5801,15 +5801,22 @@ static Model LoadM3D(const char *fileName)
                         !m3d->vertex[m3d->face[j].vertex[2]].color) vcolor = 1;
                 }
 
-                model.meshes[k].vertexCount = l*3;
+                model.meshes[k].vertexCount = l * 3;
                 model.meshes[k].triangleCount = l;
-                model.meshes[k].vertices = (float *)RL_CALLOC(model.meshes[k].vertexCount*3, sizeof(float));
-                model.meshes[k].texcoords = (float *)RL_CALLOC(model.meshes[k].vertexCount*2, sizeof(float));
-                model.meshes[k].normals = (float *)RL_CALLOC(model.meshes[k].vertexCount*3, sizeof(float));
+                model.meshes[k].vertices = (float*)RL_CALLOC(model.meshes[k].vertexCount * 3, sizeof(float));
+                model.meshes[k].texcoords = (float*)RL_CALLOC(model.meshes[k].vertexCount * 2, sizeof(float));
+                model.meshes[k].normals = (float*)RL_CALLOC(model.meshes[k].vertexCount * 3, sizeof(float));
 
                 // If no map is provided, or we have colors defined, we allocate storage for vertex colors
                 // M3D specs only consider vertex colors if no material is provided, however raylib uses both and mixes the colors
-                if ((mi == M3D_UNDEF) || vcolor) model.meshes[k].colors = RL_CALLOC(model.meshes[k].vertexCount*4, sizeof(unsigned char));
+                if ((mi == M3D_UNDEF) || vcolor) model.meshes[k].colors = RL_CALLOC(model.meshes[k].vertexCount * 4, sizeof(unsigned char));
+
+                // If no map is provided and we allocated vertex colors, set them too white
+                if ((mi == M3D_UNDEF) && model.meshes[k].colors != NULL)
+                {
+                    for (int c = 0; c < model.meshes[k].vertexCount * 4; c++)
+                        model.meshes[k].colors[c] = 255;
+                }
```

Comment:
this is the real fix, sorry about the other changes, my system auto formats to match the rest of the code.

Reply:
Ok, implemented!